linera-views/src/scylla_db.rs

```diff
@@ -715,72 +558,119 @@ impl ScyllaDbStoreInternal {
+        if !Self::is_allowed_name(namespace) {
+            return Err(ScyllaDbContextError::InvalidTableName);
+        }
         let session = SessionBuilder::new()
-            .known_node(store_config.uri.as_str())
+            .known_node(config.uri.as_str())
             .build()
             .await?;
-        let query = format!("DROP TABLE IF EXISTS kv.{};", store_config.table_name);
+        // We check the way the test can fail. It can fail in different ways.
+        let query = format!("SELECT dummy FROM kv.{} ALLOW FILTERING", namespace);
+
+        // Execute the query
+        let result = session.query(query, &[]).await;
+
+        // The missing table translates into a very specific error that we matched
+        let miss_msg1 = format!("unconfigured table {}", namespace);
+        let miss_msg1 = miss_msg1.as_str();
+        let miss_msg2 = "Undefined name dummy in selection clause";
+        let miss_msg3 = "Keyspace kv does not exist";
+        let Err(error) = result else {
+            // If ok, then the table exists
+            return Ok(true);
+        };
+        let test = match &error {
+            QueryError::DbError(db_error, msg) => {
+                if *db_error != DbError::Invalid {
+                    false
+                } else {
+                    msg.as_str() == miss_msg1
+                        || msg.as_str() == miss_msg2
+                        || msg.as_str() == miss_msg3
+                }
+            }
+            _ => false,
+        };
+        if test {
+            Ok(false)
+        } else {
+            Err(ScyllaDbContextError::ScyllaDbQueryError(error))
+        }
+    }
+
+    async fn create(config: &Self::Config, namespace: &str) -> Result<(), ScyllaDbContextError> {
+        if !Self::is_allowed_name(namespace) {
+            return Err(ScyllaDbContextError::InvalidTableName);
+        }
+        let session = SessionBuilder::new()
+            .known_node(config.uri.as_str())
+            .build()
+            .await?;
+        // Create a keyspace if it doesn't exist
+        let query = "CREATE KEYSPACE IF NOT EXISTS kv WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 }";
         session.query(query, &[]).await?;
+
+        // Create a table if it doesn't exist
+        // The schema appears too complicated for non-trivial reasons.
+        // See TODO(#1069).
+        let query = format!(
+            "CREATE TABLE kv.{} (dummy int, k blob, v blob, primary key (dummy, k))",
+            namespace
+        );
+        let _query = session.query(query, &[]).await?;
         Ok(())
     }
 
-    async fn new(
-        store_config: ScyllaDbStoreConfig,
-    ) -> Result<(Self, TableStatus), ScyllaDbContextError> {
+    async fn delete(config: &Self::Config, namespace: &str) -> Result<(), ScyllaDbContextError> {
+        if !Self::is_allowed_name(namespace) {
+            return Err(ScyllaDbContextError::InvalidTableName);
+        }
```

Comment:
Do we ever use `is_allowed_name` in another context? Otherwise I'd make it return a `Result`:

```suggestion
        Self::check_allowed_name(namespace)?;
```

Reply:
I used `check_namespace`for that.